src/types/reducers.js

```diff
@@ -39,19 +39,19 @@ import type { PathSet } from '../utils/path.js';
 export type Reducer<T> = (T | void, Action) => T;
 
 export type SymbolicationStatus = 'DONE' | 'SYMBOLICATING';
-export type ThreadViewOptions = {
-  selectedCallNodePath: CallNodePath,
-  expandedCallNodePaths: PathSet,
-  selectedMarker: IndexIntoMarkersTable | -1,
-};
+export type ThreadViewOptions = {|
+  +selectedCallNodePath: CallNodePath,
+  +expandedCallNodePaths: PathSet,
+  +selectedMarker: IndexIntoMarkersTable | -1,
+|};
 
-export type ProfileSharingStatus = {
-  sharedWithUrls: boolean,
-  sharedWithoutUrls: boolean,
-};
+export type ProfileSharingStatus = {|
+  +sharedWithUrls: boolean,
+  +sharedWithoutUrls: boolean,
+|};
 
-export type ProfileViewState = {
-  viewOptions: {
+export type ProfileViewState = {|
+  +viewOptions: {|
     perThread: ThreadViewOptions[],
```

Comment:
For some reason, making this object read only conflicts with the definition of `Reducer<...>`. I believe there's an issue in Flow, but I didn't want to upgrade Flow in this PR...